python/ray/_private/node.py

```diff
@@ -349,29 +342,31 @@ def __init__(
 
         if not connect_only:
             self.start_ray_processes()
-            # we should update the address info after the node has been started
+            # Wait for the node info to be available in the GCS so that
+            # we know it's started up.
+            node_info = None
             try:
-                ray._private.services.wait_for_node(
-                    self.gcs_address,
-                    self._plasma_store_socket_name,
-                )
+                time.sleep(0.1)  # give raylet some time to register
```

Comment:
```suggestion
                # Grace period to let the Raylet register with the GCS.
                # We retry in a loop in case it takes longer than expected.
                time.sleep(0.1)
```